src/main/java/org/gridsuite/odre/server/utils/GeographicDataParser.java

```diff
@@ -82,7 +82,34 @@ private GeographicDataParser() {
         return substations;
     }
 
-    public static Map<String, LineGeoData> parseLines(BufferedReader aerialLinesBr, BufferedReader undergroundLinesBr) {
+    private static double distanceCoordinate(Coordinate coord1, Coordinate coord2) {
+        return DistanceCalculator.distance(coord1.getLat(), coord1.getLon(), coord2.getLat(), coord2.getLon());
+    }
+
+    public static String findSubstationStart(Map<String, SubstationGeoData> substationGeoData, String lineId, List<Coordinate> coordinates) {
+        String substation1 = lineId.substring(0, 5).trim();
+        String substation2 = lineId.substring(8).trim();
+        SubstationGeoData geo1 = substationGeoData.get(substation1);
+        SubstationGeoData geo2 = substationGeoData.get(substation2);
+
+        if (geo1 == null && geo2 == null) {
+            return "";
+        } else if (geo1 != null && geo2 != null) {
+            return distanceCoordinate(geo1.getCoordinate(), coordinates.get(0))
```

Comment:
Instead of checking 
d(geo1, coord(0)) < d(geo2, coord(0))
isn't it better to check d(geo1, coord(0)) < d(geo1, coord(n)) ?
imagine we only have the data for a few pylons, next to geo2. then with your code the result is always geo2


In fact, maybe we should do more checks:
d(geo1, coord(0)) < d(geo2, coord(0))
d(geo2, coord(n)) < d(geo1, coord(n))
d(geo1, coord(0)) < d(geo1, coord(n))
d(geo2, coord(n)) < d(geo2, coord(0))

and return an error or something if it's not the case ?